supervision/dataset/formats/yolo.py

```diff
@@ -86,30 +87,47 @@ def yolo_annotations_to_detections(
             if with_masks:
                 relative_polygon.append(_box_to_polygon(box=box))
         elif len(values) > 5:
-            polygon = _parse_polygon(values=values[1:])
-            relative_xyxy.append(polygon_to_xyxy(polygon=polygon))
-            if with_masks:
-                relative_polygon.append(polygon)
+            if is_obb:
+                polygon = _parse_polygon(values=values[1:])
+                relative_xyxyxyxy.append(np.array(values[1:]))
+                if with_masks:
+                    relative_polygon.append(polygon)
+            else:
+                polygon = _parse_polygon(values=values[1:])
+                relative_xyxy.append(polygon_to_xyxy(polygon=polygon))
+                if with_masks:
+                    relative_polygon.append(polygon)
```

Comment:
If we're doing the same operations inside both `if` branches, let's move them outside.

```
polygon = _parse_polygon(values=values[1:])
if with_masks:
        relative_polygon.append(polygon)
```

can be moved above `if is_obb`

Reply:
Okay, will do.